src/screens/ConfigurePMTs/PaymentMethodConfig.res

```diff
@@ -0,0 +1,214 @@
+  let (countries, setCountries) = React.useState(_ => [])
+
+  let {
+    payment_method_index,
+    payment_method_types_index,
+    connector_name,
+    payment_method_type,
+    merchant_connector_id,
+  } = paymentMethodConfig
+  open APIUtils
+  let connectorList = HyperswitchAtom.connectorListAtom->Recoil.useRecoilValueFromAtom
+  let fetchDetails = useGetMethod()
+  let updateDetails = useUpdateMethod()
+  let showToast = ToastState.useShowToast()
+
+  let getProcessorDetails = async () => {
+    open LogicUtils
+    try {
+      let paymentMethoConfigUrl = getURL(~entityName=PAYMENT_METHOD_CONFIG, ~methodType=Get, ())
+      let data =
+        connectorList
+        ->Array.filter(item =>
+          item.merchant_connector_id === paymentMethodConfig.merchant_connector_id
+        )
+        ->Array.get(0)
+        ->Option.getOr(Dict.make()->ConnectorListMapper.getProcessorPayloadType)
+      let encodeConnectorPayload = data->PaymentMethodConfigUtils.encodeConnectorPayload
+      let res = await fetchDetails(
+        `${paymentMethoConfigUrl}?connector=${connector_name}&paymentMethodType=${payment_method_type}`,
+      )
+      let countries =
+        res
+        ->getDictFromJsonObject
+        ->getArrayFromDict("countries", [])
+        ->Array.map(item => {
+          let dict = item->getDictFromJsonObject
+          let a: SelectBox.dropdownOption = {
+            label: dict->getString("name", ""),
+            value: dict->getString("code", ""),
+          }
+          a
+        })
+      let currencies =
+        res
+        ->getDictFromJsonObject
+        ->getArrayFromDict("currencies", [])
+        ->Array.map(item => {
+          let a: SelectBox.dropdownOption = {
+            label: item->getStringFromJson(""),
+            value: item->getStringFromJson(""),
+          }
+          a
+        })
+      setCountries(_ => countries)
+      setCurrencies(_ => currencies)
+
+      setInitialValues(_ => encodeConnectorPayload)
+      setShowPaymentMthdConfigModal(_ => true)
+    } catch {
+    | _ => setShowPaymentMthdConfigModal(_ => false)
+    }
+  }
+
+  let onSubmit = async (values, _) => {
+    open LogicUtils
+    try {
+      let url = getURL(~entityName=CONNECTOR, ~methodType=Post, ~id=Some(merchant_connector_id), ())
+      let _ = await updateDetails(url, values, Post, ())
+      let _ = await setReferesh()
+      setShowPaymentMthdConfigModal(_ => false)
+    } catch {
+    | Exn.Error(e) => {
+        let err = Exn.message(e)->Option.getOr("Something went wrong")
+        let errorMessage = err->safeParse->getDictFromJsonObject->getString("message", "")
+        showToast(~message=errorMessage, ~toastType=ToastError, ())
+        setShowPaymentMthdConfigModal(_ => true)
+      }
+    }
+    Nullable.null
+  }
+  let id = `payment_methods_enabled[${payment_method_index->Int.toString}].payment_method_types[${payment_method_types_index->Int.toString}]`
```

Comment:
Do we want this id to be changed everytime? As this value is let so it will be giving the values to default start on every render either we can put a useMemo or make it as a state

Reply:
yes we wanted to change the id for every render and mainly its just id used for form component